ivadomed/loader/mri2d_segmentation_dataset.py

```diff
@@ -278,3 +323,26 @@ def __getitem__(self, index):
             data_dict = dropout_input(data_dict)
 
         return data_dict
+
+    def determine_cache_need(self, item: tuple, n_slice: int):
+        """
+        When Cache flag is not explicitly set, determine whether to cache the data or not
+        Args:
+            item: an EXAMPLE, typical Tuple structure contain the main data.
+            n_slice: number of slice in one file_name_pairs.
+
+        Returns:
+
+        """
+        size_item_in_bytes = get_obj_size(item)
+
+        # Size limit: 4GB GPU RAM, keep in mind tranform etc might take MORE!
```

Comment:
Why not fetching the RAM available (Python lib can do that) and use it instead?

Reply:
This has been implemented. It would probably be nicer to use `psutil` toolbox but for the sake of simplifying dependencies, a few system specific scripts are added to `util.py` obtain these information.